### DISCUSSION TITLE
Can someone tell me what is causing this animation to behave like this on iOS and iPadOS.

### DISCUSSION THREAD

**Discussion by paperpluto** [tailwindlabs/tailwindcss#11961] [Help]
**What version of Tailwind CSS are you using?**
v 3.3.3

**What build tool (or framework if it abstracts the build tool) are you using?**
Next.js 13.4.19

**What version of Node.js are you using?**
v 19.7.0

**What browser are you using?**
Chrome, Safari

**What operating system are you using?**
Windows, iOS, iPadOS

**Reproduction URL**
https://codesandbox.io/p/sandbox/confident-brown-4p8y7l

https://4p8y7l-5173.csb.app/

**Describe your issue**
There is some weird glitch or something that is happening when I click on the toggle button "loading: true". The loading bug occurs only on iPad, iOS, (not tested on MacOS). It works fine on Windows and Android browsers. I teseted on Chrome and Safari on iOS and iPadOS and Chrome on Windows and Android.

**Comment by RobinMalfait** [comment]
Hey!

This is also happening in Safari on desktop. I think what's happening is that the moment you "pause" the animation then Safari (and iOS) will literally pause at the current spot. Then you add `w-full` which makes the loading spinner larger from wherever the animation is in the cycle.

On other browsers like Chrome on desktop, a "pause" is more like a "pause the animation, and go back to 0%"

One way of solving this is to use `animate-none` to completely stop the animation which also resets its position: https://codesandbox.io/p/sandbox/mystifying-fast-n2336q?file=%2Fsrc%2FApp.css%3A7%2C38

**Reply by paperpluto** [reply]
Can you please tell me why the animation stays paused after re-running (setting the loading true) in iOS Safari and Chrome. It should animate.

**Reply by RobinMalfait** [reply]
This happens because the browser is batching updates which results in the animation not playing anymore. This StackOverflow post summarises it: https://stackoverflow.com/a/30072037

One fun thing you can do in React is this step: "Clone the element, remove it, and insert the clone"

You can achieve that by setting a `key` prop to the div. This will change the identity of the div and React will completely destroy the div and re-create it. Resulting in the "Clone the element, remove it, and insert the clone" scenario.

In the inner most div, add: `key={loading}`. I've update the CodeSandbox from above: https://codesandbox.io/p/sandbox/mystifying-fast-n2336q?file=/src/App.tsx:29,1-30,1

Hope this helps!
